File: dart-client/lib/model/installation_add_object.dart
part of swagger.api;

class InstallationAddObject {
  
  String deviceToken = null;
  

  String deviceType = null;
  

  List<String> channels = [];
  
  InstallationAddObject();

  @override
  String toString() {
    return 'InstallationAddObject[deviceToken=$deviceToken, deviceType=$deviceType, channels=$channels, ]';
  }

  InstallationAddObject.fromJson(Map<String, dynamic> json) {
    if (json == null) return;
    deviceToken =
        json['deviceToken']
    ;
    deviceType =
        json['deviceType']
    ;
    channels =
        (json['channels'] as List).map((item) => item as String).toList()
    ;
  }

  Map<String, dynamic> toJson() {
    return {
      'deviceToken': deviceToken,
      'deviceType': deviceType,
      'channels': channels
     };
  }

  static List<InstallationAddObject> listFromJson(List<dynamic> json) {
    return json == null ? new List<InstallationAddObject>() : json.map((value) => new InstallationAddObject.fromJson(value)).toList();
  }

  static Map<String, InstallationAddObject> mapFromJson(Map<String, Map<String, dynamic>> json) {
    var map = new Map<String, InstallationAddObject>();
    if (json != null && json.length > 0) {
      json.forEach((String key, Map<String, dynamic> value) => map[key] = new InstallationAddObject.fromJson(value));
    }
    return map;
  }
}

File: csharp-dotnet2-client/src/main/CsharpDotNet2/IO/Swagger/Model/InstallationAddObject.cs
using System;
using System.Text;
using System.Collections;
using System.Collections.Generic;
using System.Runtime.Serialization;
using Newtonsoft.Json;

namespace IO.Swagger.Model {

  /// <summary>
  /// 
  /// </summary>
  [DataContract]
  public class InstallationAddObject : Dictionary<String, string> {
    /// <summary>
    /// Gets or Sets DeviceToken
    /// </summary>
    [DataMember(Name="deviceToken", EmitDefaultValue=false)]
    [JsonProperty(PropertyName = "deviceToken")]
    public string DeviceToken { get; set; }

    /// <summary>
    /// Gets or Sets DeviceType
    /// </summary>
    [DataMember(Name="deviceType", EmitDefaultValue=false)]
    [JsonProperty(PropertyName = "deviceType")]
    public string DeviceType { get; set; }

    /// <summary>
    /// Gets or Sets Channels
    /// </summary>
    [DataMember(Name="channels", EmitDefaultValue=false)]
    [JsonProperty(PropertyName = "channels")]
    public List<string> Channels { get; set; }


    /// <summary>
    /// Get the string presentation of the object
    /// </summary>
    /// <returns>String presentation of the object</returns>
    public override string ToString()  {
      var sb = new StringBuilder();
      sb.Append("class InstallationAddObject {\n");
      sb.Append("  DeviceToken: ").Append(DeviceToken).Append("\n");
      sb.Append("  DeviceType: ").Append(DeviceType).Append("\n");
      sb.Append("  Channels: ").Append(Channels).Append("\n");
      sb.Append("}\n");
      return sb.ToString();
    }

    /// <summary>
    /// Get the JSON string presentation of the object
    /// </summary>
    /// <returns>JSON string presentation of the object</returns>
    public  new string ToJson() {
      return JsonConvert.SerializeObject(this, Formatting.Indented);
    }

}
}
